src/spikeinterface/comparison/comparisontools.py

```diff
@@ -3,12 +3,13 @@
 """
 
 from __future__ import annotations
+from typing import Iterable
 
 
 import numpy as np
 
 
-def count_matching_events(times1, times2, delta=10):
+def count_matching_events(times1: Iterable, times2: Iterable, delta: int = 10):
```

Comment:
As @h-mayorquin has pointed out to many many many times (because I make the same mistake), is that `str` is also an `Iterable` so in this case this type hint would be too permissive. We've discussed this before that it `ArrayLike` is probably the best type, but I personally don't like it because I don't think it is beginner friendly. I think this is an edge case, could you remind me based on your reading of this code: is this function public or private? If private I would probably just give it a type of np.ndarray or list since that is likely what it users internally. If public then this is a debate between doing 
```
list | np.ndarray | tuple
```
vs
```
ArrayLike
```
which I think actually comes from numpy.typing.

Reply:
Thanks for pointing that out! I didn't even consider the overly permissiveness of `Iterable`. Since this function is private (can't find a reference to it on readthedocs), I will go ahead with just giving it a np.ndarray. 

This function annotation is something that I put since i was starting to annotate everything before realizing that annotating the return types is more useful than the inputs. I can also just remove the hint since it is not necessary.

Reply:
@tabedzki has this been updated?

Reply:
Has been now to reflect zm711's suggestion for simple internal type hint